bin/systemml-spark-submit.py

```diff
@@ -18,40 +18,42 @@
-    this_script = sys.argv[0]
-    print('Usage: ' + this_script + ' <dml-filename> [arguments]')
-    sys.exit(1)
-
 if environ.get('SPARK_HOME') is None:
     print('SPARK_HOME not set')
 
-if len(sys.argv) < 2:
-    print('Wrong usage')
-    print_usage_and_exit()
-
-parser = argparse.ArgumentParser(description='System-ML Spark Submit Script', add_help=False, formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-parser.add_argument('--help', action='help', help='Print this usage message and exit')
-parser.add_argument('--master', default='local[*]', help='local, yarn-client, yarn-cluster', metavar='\b')
-parser.add_argument('--driver-memory', default='5G', help='Memory for driver (e.g. 512M)', metavar='\b')
-parser.add_argument('--num-executors', default='2', help='Number of executors to launch', metavar='\b')
-parser.add_argument('--executor-memory', default='2G', help='Memory per executor', metavar='\b')
-parser.add_argument('--executor-cores', default='1', help='Number of cores', metavar='\b')
-parser.add_argument('--conf', default='', help='Configuration settings', metavar='\b')
-parser.add_argument('-f', required=True, help='DML script file name', metavar='\b')
-parser.add_argument('-nvargs', nargs='*')
-args = parser.parse_args()
-arg_dict = vars(args)
+cparser = argparse.ArgumentParser(description='System-ML Spark Submit Script', add_help=False)
+cparser.add_argument('--help', action='help', help='Print this usage message and exit')
+
+# SPARK-SUBMIT Options
+cparser.add_argument('--master', default='local[*]', help='local, yarn-client, yarn-cluster', metavar='')
+cparser.add_argument('--driver-memory', default='5G', help='Memory for driver (e.g. 512M)', metavar='')
+cparser.add_argument('--num-executors', default='2', help='Number of executors to launch', metavar='')
+cparser.add_argument('--executor-memory', default='2G', help='Memory per executor', metavar='')
+cparser.add_argument('--executor-cores', default='1', help='Number of cores', metavar='')
+cparser.add_argument('--conf', default='', help='Spark configuration file', nargs='+', metavar='')
+
+# SYSTEM-ML Options
+cparser.add_argument('-nvargs', required=True, help='List of attributeName-attributeValue pairs', nargs='+')
+cparser.add_argument('-args', help='List of positional argument values', metavar='', nargs='+')
+cparser.add_argument('-config', help='System-ML configuration file (e.g SystemML-config.xml)', metavar='')
+cparser.add_argument('-stats',  default='10', help='Monitor and report caching/recompilation statistics, '
+                                                   'heavy hitter <count> is 10 unless overridden')
+cparser.add_argument('-explain', default='runtime', help='explains plan levels can be hops, runtime, '
+                                                         'recompile_hops, recompile_runtime', metavar='')
+cparser.add_argument('-exe', default='hybrid_spark', help='System-ML backend (e.g spark, spark-hybrid)', metavar='')
```

Comment:
In line number `126` I have to make a dictionary call `args.exe`. Unfortunately `args.exec` seems to give me a problem.

![image](https://cloud.githubusercontent.com/assets/4648995/26279861/9f5ed514-3d8d-11e7-8b8c-1511d9c76c5e.png)